### PR TITLE
coordinator: fix TestRecovery overrides

### PR DESCRIPTION
**Pull request by burgerdev**
While refactoring the userapi, I found out that the overrides for TestRecovery requests are broken. The intention was to override the request data with what's in the test case struct, but this did not work as expected for empty seed and empty salt. We did not detect this because the error message was unchecked and we got the expected `InvalidArgument`, but from the latest transition signature check and not from the seedengine constructor.

### CODE REVIEW

File: coordinator/internal/authority/userapi_test.go [hunk 1]
```diff
@@ -253,32 +253,35 @@ func TestRecovery(t *testing.T) {
...
+		salt        *[]byte
+		wantCode    codes.Code
+		wantMessage string
 	}{
 		{
-			name:     "empty seed",
-			salt:     salt[:],
-			wantCode: codes.InvalidArgument,
+			name:        "empty seed",
+			seed:        toPtr[[]byte](nil),
```
Comment (3u13r) [inline coordinator/internal/authority/userapi_test.go:264]: The `toPtr` is superfluous here, since type of the member variable should make nil already to a typed nil.

File: coordinator/internal/authority/userapi_test.go [hunk 2]
```diff
@@ -253,32 +253,35 @@ func TestRecovery(t *testing.T) {
 	var seed [32]byte
 	var salt [32]byte
 	testCases := []struct {
-		name     string
-		seed     []byte
-		salt     []byte
-		wantCode codes.Code
+		name        string
+		seed        *[]byte
+		salt        *[]byte
```
Comment (3u13r) [inline coordinator/internal/authority/userapi_test.go:258]: Since `nil` is a valid slice, why not use `[]byte` as the type. The default initialization is also nil:
```go
package main

import "fmt"

func main() {
	var s []int

	fmt.Println(s==nil) // prints true
}
```

**Review by 3u13r — CHANGES_REQUESTED** [review]

File: coordinator/internal/authority/userapi_test.go (hunk 2 above)
Comment (burgerdev) [reply coordinator/internal/authority/userapi_test.go:258]: The idea is to decide whether to override the value in the test body based on this pointer being set or nil, and I want to be able to override to nil.

File: coordinator/internal/authority/userapi_test.go (hunk 1 above)
Comment (burgerdev) [reply coordinator/internal/authority/userapi_test.go:264]: There's a difference: I'm creating a non-nil pointer to a nil slice here, which makes the test body override the seed with a nil slice.

Comment (3u13r) [reply coordinator/internal/authority/userapi_test.go:264]: Makes sense, but I think there's still another way of testing the emptiness test cases with initialized 0 length slices and then using the nil slice to decide whether to replace or not. If you want to explicitly test nil inputs then I'd create different test cases (one with nil and one with zero length slices), if not then I'd go with just zero length slices.

Comment (burgerdev) [reply coordinator/internal/authority/userapi_test.go:264]: I explicitly want to test with nil slices, there is a test with a short slice already.

Comment (3u13r) [reply coordinator/internal/authority/userapi_test.go:264]: Alright, good point

**Review by 3u13r — APPROVED** [review]
